src/launchoptions.rs

```diff
@@ -0,0 +1,211 @@
+/// -conclearlog clears the console.log file on launch of TF2
+/// -usercon allows local RCON to be used
+/// -g15 enables Logitech G15 keyboard support (used for the console command `g15_dumpplayer`)
+pub const TF2_REQUIRED_OPTS: [&str; 4] = ["-condebug", "-conclearlog", "-usercon", "-g15"];
+
+/// Handles referencing the VDF store of a Steam app's launch options and provides an interface to read
+/// and write launch options based on a set of required options.
+pub struct LaunchOptionsV2 {
+    local_config: PathBuf,
+    launch_args_regex: Regex,
+    app_data: Option<String>,
+    new_app_data: Option<String>,
+}
+
+/// Read the local steam library folders for data (stored in VDF/KeyValues format) on the configured launch options for the give app
+/// ID. If necessary, warn about the missing launch options.
+impl LaunchOptionsV2 {
+    /// Get the Launch options of the given app for the given steam user (should be the user provided in loginusers.vdf)
+    /// This can fail if the `localconfig.vdf` file is not present in the expected path, or if the given app is not present
+    /// in the users list of launched apps (either they don't own it or have never played it).
+    ///
+    /// The Source VDF file format is just a wrapped 'KeyValues' format, which is essentially the cursed brother of JSON.
+    /// This format separates keys and values via '\t\t' rather than ':', and is (obviously) white space sensitive. This
+    /// allows us to make the guarenteed assumption that for as long as the specification of `localconfig.vdf` does not change,
+    /// the app block we are looking for will start with `\t\t\t\t\t"<app id>"`, and as such all of the required app data is at
+    /// least 6 tabs deep, and the closing '}' character is the very next '}' found only 5 tabs deep.
+    pub fn new(user: SteamID, target_app: String) -> Result<LaunchOptionsV2, anyhow::Error> {
+        let span = tracing::span!(Level::INFO, "LaunchOptions");
+        let _enter = span.enter();
+
+        let config_path: PathBuf = locate_steam_launch_configs(user)
+            .context("Failed to find localconfig.vdf in Steam dir.")?;
+        let mut data: Vec<u8> = Vec::new();
+        tracing::debug!("Found Steam local app config.");
+
+        if let Ok(mut f) = File::open(&config_path) {
+            f.read_to_end(&mut data)
+                .context(format!("Failed to read {:?}", config_path))?;
+        }
+        let binding = &String::from_utf8_lossy(&data);
+
+        let apps_regex =
+            Regex::new(r#"\t{4}"[aA]pps"([\s\S]+)\t{5}}"#).expect("Apps regex construction");
+
+        let caps = apps_regex
+            .captures(binding)
+            .context("No capture groups found -  no apps list present in localconfig.vdf.")?;
+
+        let mut matched_app_block: Option<String> = None;
+        let open_match = format!("\t\t\t\t\t\"{}\"", target_app);
+
+        let mat_opt = caps.get(0);
```

Comment:
According to [the documentation](https://docs.rs/regex/latest/regex/struct.Captures.html):
> The group at index 0 corresponds to the overall match. It is always present in every match and never has a name.

So if you're trying to access the capture group in your regex above you'll want to be looking at index 1.

Reply:
Yis, but i mistakenly remembered that as a previous iteration of the regex which had no cap groups. good catch